jenkins/github_helpers.py

```diff
@@ -157,6 +157,8 @@ def close_existing_pull_requests(self, repository, user_login, user_name, target
             user = pr.user
             if user.login == user_login and user.name == user_name and pr.base.ref == target_branch:
                 branch_name = pr.head.ref
+                if self.is_cleanup_pr(branch_name):
+                    continue
```

Comment:
This shouldn't be needed, since there's now a command line option to control whether this is called:

https://github.com/edx/testeng-ci/blob/0c83251df0ebab50400b266d3b9f9916502405fe/jenkins/pull_request_creator.py#L123

Are you seeing cleanup job runs causing PR closes and branch deletions? If so, we should look into why that option isn't getting passed properly.

Reply:
Here's where that option is getting added:

https://github.com/edx/jenkins-job-dsl/blob/07a45f79063dd44fe21fcbd9222238d3174e9dea/testeng/resources/cleanup-python-code-create-pr.sh#L69

Reply:
we still want to close any existing requirement upgrade PRs but leave the cleanup PRs, this command line argument is being passed by the cleanup script, we don't want to pass this from the requirements upgrade job and still need to make sure it doesn't close any existing cleanup PRs
one possible solution would be to generate a new token for cleanup PR creation, currently, it is utilizing `edx-requirements-bot`'s  credentials, it checks if there are any existing PRs created by the same user and it then tries to close those ones

Reply:
It sounds like you're saying that new upgrade PRs are closing cleanup PRs. Can you confirm whether that's what prompted this?

If so, it sounds like what we need is filtering on the branch name -- pull_request_creator.py should pass `base_branch_name` through to `close_existing_pull_requests` as a new optional parameter, and that should be used for filtering instead of `is_cleanup_pr`. That will give us much better control.

Reply:
yes, that's what is happening. `--no-delete-old-pull-requests` parameter is being passed by [cleanup job script](https://github.com/edx/jenkins-job-dsl/blob/master/testeng/resources/cleanup-python-code-create-pr.sh) but the [upgrade PRs script](https://github.com/edx/jenkins-job-dsl/blob/master/testeng/resources/upgrade-python-requirements.sh) has `--delete-old-pull-requests` parameter specified that causes new upgrade PRs to close exiting cleanup PRs

Reply:
we could have this check added here to avoid closing those PRs or get new credentials for cleanup PRs (i.e a new bot for creating cleanup PRs)

Reply:
now that we've started moving upgrade jobs to use Github workflows instead this should get fixed but in the meantime what do you suggest should be the solution? @timmc-edx @jmbowman

Reply:
I _think_ this will do what we want: https://github.com/edx/testeng-ci/pull/255

Reply:
I've merged and tested it, and I think it works as needed.

Reply:
we don't need this PR, right?

Reply:
That's right, we can close this one now. 

There's actually one more fix we _might_ want to make in this space, which is that you currently can't create two cleanup PRs for the same target commit. But that seems lower priority.